engine/src/test/scala/cromwell/engine/workflow/workflowstore/SqlWorkflowStoreSpec.scala

```diff
@@ -244,22 +253,128 @@ class SqlWorkflowStoreSpec extends AnyFlatSpec with CromwellTimeoutSpec with Mat
-        _ = startableWorkflows.map(_.id).intersect(submissionResponsesIncluded2.map(_.id).toList).size should be(1)
-        _ = startableWorkflows.map(_.id).intersect(submissionResponsesIncluded3.map(_.id).toList).size should be(1)
-        _ = startableWorkflows.map(_.id).size should be(3)
-        _ <- workflowStore.deleteFromStore(submissionResponsesExcluded.head.id) // Tidy up
-        _ <- workflowStore.deleteFromStore(submissionResponsesIncluded1.head.id) // Tidy up
-        _ <- workflowStore.deleteFromStore(submissionResponsesIncluded2.head.id) // Tidy up
-        _ <- workflowStore.deleteFromStore(submissionResponsesIncluded3.head.id) // Tidy up
+        // first submission of 10 workflows for hogGroup "Goldfinger"
+        goldFingerSubmissions <- Future.sequence(for (_ <- 1 to 10) yield workflowStore.add(includedGroupSourceFilesCollection1))
+        goldFingerWorkflowIds = goldFingerSubmissions.flatMap(_.map(_.id).toList)
+
+        // second submission of 10 workflows for hogGroup "Highlander"
+        highlanderSubmissions <- Future.sequence(for (_ <- 1 to 15) yield workflowStore.add(includedGroupSourceFilesCollection2))
+        highlanderWorkflowIds = highlanderSubmissions.flatMap(_.map(_.id).toList)
+
+        // since both hog groups have 0 workflows running, the hog groups are sorted alphabetically and first one is picked
+        startableWorkflows1 <- workflowStore.fetchStartableWorkflows(5, "A08", 5.minutes, excludedGroups = Set.empty[String])
+        _ = startableWorkflows1.map(_.hogGroup.value).toSet.head should be("Goldfinger")
+        _ = startableWorkflows1.map(_.id).foreach(x => goldFingerWorkflowIds.toList should contain(x))
+        _ = updateWfToRunning(startableWorkflows1)
+
+        startableWorkflows2 <- workflowStore.fetchStartableWorkflows(5, "A08", 5.minutes, excludedGroups = Set.empty[String])
+        _ = startableWorkflows2.map(_.hogGroup.value).toSet.head should be("Highlander")
+        _ = startableWorkflows2.map(_.id).foreach(x => highlanderWorkflowIds.toList should contain(x))
+        _ = updateWfToRunning(startableWorkflows2)
+
+        // new submission for hog group "Finding Forrester"
+        foresterSubmissions <- Future.sequence(for (_ <- 1 to 10) yield workflowStore.add(includedGroupSourceFilesCollection3))
+        foresterWorkflowIds = foresterSubmissions.flatMap(_.map(_.id).toList)
+
+        // now hog group "Finding Forrester" has 0 workflows running, hence it is picked to run
+        startableWorkflows3 <- workflowStore.fetchStartableWorkflows(5, "A08", 5.minutes, excludedGroups = Set.empty[String])
+        _ = startableWorkflows3.map(_.hogGroup.value).toSet.head should be("Finding Forrester")
+        _ = startableWorkflows3.map(_.id).foreach(x => foresterWorkflowIds.toList should contain(x))
+        _ = updateWfToRunning(startableWorkflows3)
+
+        // since all 3 hog groups have 5 workflows running each, the hog groups are sorted alphabetically and first one is picked
```

Comment:
Once this order has been determined, is it unchanged until all of those hog groups are finished? i.e., Currently 5 hog groups waiting, they're sorted, then they all run. If there are more groups following this, they do the same as the previous.

Reply:
The alphabetical sorting only comes into play if there is more than 1 hog group that have same count of actively running workflows. In this case, we alphabetically sort the hog groups and pick the first hog group to run workflows. 
> Once this order has been determined, is it unchanged until all of those hog groups are finished?

The order is not fixed. It is determined every time we fetch new workflows to start. I think [this test case](https://github.com/broadinstitute/cromwell/blob/93fb0c6601beef467cd596dc2468354e4d54dc5d/engine/src/test/scala/cromwell/engine/workflow/workflowstore/SqlWorkflowStoreSpec.scala#L323-L379) is a good example of this.